lib/AST/ASTPrinter.cpp

```diff
@@ -5350,23 +5538,37 @@ void PrintAST::visitDestructureTupleExpr(DestructureTupleExpr *expr) {
 }
 
 void PrintAST::visitDynamicMemberRefExpr(DynamicMemberRefExpr *expr) {
+  visit(expr->getBase());
+  Printer << ".";
+  Printer << expr->getMember().getDecl()->getName();
 }
 
 void PrintAST::visitDynamicSubscriptExpr(DynamicSubscriptExpr *expr) {
+  visit(expr->getBase());
+  printArgumentList(expr->getArgs()->getOriginalArgs(), true);
 }
 
 void PrintAST::visitPointerToPointerExpr(PointerToPointerExpr *expr) {
   visit(expr->getSubExpr());
 }
 
 void PrintAST::visitUnresolvedMemberExpr(UnresolvedMemberExpr *expr) {
+  Printer << ".";
+  Printer << expr->getName();
 }
 
 void PrintAST::visitDiscardAssignmentExpr(DiscardAssignmentExpr *expr) {
   Printer << "_";
 }
 
 void PrintAST::visitEditorPlaceholderExpr(EditorPlaceholderExpr *expr) {
+  if (expr->isTypeReference()) {
+    printTypeLoc(expr->getPlaceholderTypeRepr());
+  } else {
+    Printer << "<#";
+    Printer << expr->getPlaceholder();
+    Printer << "#>";
+  }
```

Comment:
I actually think you only need to print `getPlaceholder()` here (for both cases) since that should include the entire `<#...#>` string